test/Controllers/ResourcesTest.php

```diff
@@ -0,0 +1,234 @@
+use GonebusyLib\Models\UpdateResourceByIdBody;
+// use GonebusyLib\Controllers\ResourcesController;
+use GonebusyLib\Models\GenderEnum;
+
+class ResourcesTest extends TestCase
+{
+    /**
+     * To contain the GonebusyLib\GonebusyClient
+     */
+    protected $client;
+
+    /**
+     * To contain the GonebusyLib\Controllers\ResourcesController
+     */
+    protected $resources;
+
+    /**
+     * Create the GonebusyClient and ResourceController for each test.
+     */
+    public function setUp() {
+        $this->client = new GonebusyClient();
+        $this->resources = $this->client->getResources();
+    }
+
+
+    /**
+     * Returns arbitrary resource data.
+     * @param  string $type Should be 'create' or 'update'.
+     * @return  CreateResourceBody or UpdateResourceByIdBody object with unique data to send to API
+     */
+    private function resourceBody($type) {
+        $rand = rand();
+        switch($type) {
+            case 'create':
+                return new CreateResourceBody(
+                    "name",
+                    "Staff",
+                    NULL, // capacity
+                    "description",
+                    GenderEnum::F, // gender
+                    NULL, // thing_type_id
+                    NULL // user_id detauls to self
+                );
+            case 'update':
+                return new UpdateResourceByIdBody(
+                    NULL, // capacity
+                    "another description",
+                    GenderEnum::M, // gender change
+                    "another name",
+                    NULL // thing_type_id
+                );
+        }
+    }
+
+    /**
+     * Generate data from Resource response.
+     * @param  GonebusyLib\Models\EntitiesResourceResponse $response
+     * @param  string $type Should be 'create' or 'update'.
+     * @return  CreateResourceBody or UpdateResourceByIdBody object with data from $response
+     */
+    private function bodyFromResponse($response, $type) {
+        switch($type) {
+            case 'create':
+                return new CreateResourceBody(
+                    $response->resource->name,
+                    $response->resource->resourceType,
+                    $response->resource->capacity,
+                    $response->resource->description,
+                    GenderEnum::F, //$response->resource->gender, // XXX Hardcoded as API isn't returning a value
+                    $response->resource->thingTypeId,
+                    NULL // $response->resource->ownerId
+                );
+            case 'update':
+                return new UpdateResourceByIdBody(
+                    $response->resource->capacity,
+                    $response->resource->description,
+                    GenderEnum::M, //$response->resource->gender, // XXX Hardcoded as API isn't returning a value
+                    $response->resource->name,
+                    $response->resource->thingTypeId
+                );
```

Comment:
@alexagranov if you see the `XXX` codetags above, this means the API isn't returning `gender` neither after a POST /resources/new nor PUT /resources/{id}, not even on GET /resources/{id} so I had to hardcode the mocked response to match the data I sent in the first place.

![image](https://cloud.githubusercontent.com/assets/1477535/23935870/06035774-0925-11e7-8249-8904ea02e71b.png)

![image](https://cloud.githubusercontent.com/assets/1477535/23935885/1c43d13a-0925-11e7-94b5-978011c299bc.png)